cmd/zpool/zpool_main.c

```diff
@@ -2462,7 +2462,14 @@ print_status_config(zpool_handle_t *zhp, status_cbdata_t *cb, const char *name,
 	(void) nvlist_lookup_uint64_array(root, ZPOOL_CONFIG_SCAN_STATS,
 	    (uint64_t **)&ps, &c);
 
-	if (ps != NULL && ps->pss_state == DSS_SCANNING && children == 0) {
+	/*
+	 * If you force fault a drive that's resilvering, it's scan stats can
+	 * get frozen in time, giving the false impression that it's
+	 * being resilvered.  That's why check the state to see if the vdev
+	 * is healthy before reporting "resilvering" or "repairing".
+	 */
+	if (ps != NULL && ps->pss_state == DSS_SCANNING && children == 0 &&
+	    vs->vs_state == VDEV_STATE_HEALTHY) {
```

Comment:
You'll want to add the same check for `vs->vs_rebuild_processed` below.  This is the counter which is incremented for the sequential resilver, `zpool replace -s`.